docs/jetstream.md

```diff
@@ -11,39 +11,107 @@ There are two components to this implementation:
 
 ### Controller
 
-The controller reconciles `NatsJetStreamChannel` CRDs by checking the availability of the `jetstream-channel-dispatcher`
-and ensures a `Service` corresponding to the channel exists with an `externalName` pointing to the 
-`jetstream-channel-dispatcher`.
+The controller reconciles `NatsJetStreamChannel` CRDs by ensuring the `Deployment` named `jetstream-ch-dispatcher`
+exists and ensures a `Service` corresponding to the channel exists with an `externalName` pointing to the
+`jetstream-ch-dispatcher`.
+
+#### Scoping
+
+`NatsJetStreamChannel` resources may be scoped to a namespace by adding the annotation
+`eventing.knative.dev/scope: namespace`, allowing the use of dedicated dispatcher deployments per namespace. This has
```

Comment:
I was wondering if this scope is necessary? If we have the `namespaceDefaults` for `config-ch-jetstream-defaults`

Reply:
So I've followed the Kafka implementation here:

- If the `NatsJetStreamChannel` does not have any annotation, then all of them are handled by the `knative-eventing/jetstream-ch-dispatcher` deployment, regardless of which namespace the channel exists in.
- If it does have the annotation, then a dispatcher running in the same namespace is used, using the config map defaults as discussed above.

I was wondering instead of this approach, whether to have have a flag on the controller executable like `--enable-namespace-scope`. If omitted (the default behaviour), then all channels are backed by the `knative-eventing` dispatcher. If included, then all channels are backed by a per-namespace dispatcher, and no dispatcher exists in `knative-eventing` (unless someone created a channel in that namespace)

I chose the former since this is how Kafka works and most people are going to be comparing this component to kafka.

Reply:
that's ok